src/snapred/backend/dao/state/GroupingMap.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from pathlib import Path
+from typing import Any, ClassVar, Dict, List
+
+from pydantic import BaseModel, Field, root_validator, validator
+
+from snapred.backend.dao.ObjectSHA import ObjectSHA
+from snapred.backend.dao.state.FocusGroup import FocusGroup
+from snapred.backend.log.logger import snapredLogger
+from snapred.meta.Config import Config
+
+logger = snapredLogger.getLogger(__name__)
+
+
+class GroupingMap(BaseModel):
+    # State id associated with the _default_ grouping map, at 'instrument.calibration.powder.grouping.home':
+    defaultStateId: ClassVar[ObjectSHA] = ObjectSHA(hex="aabbbcccdddeeeff", decodedKey=None)
+
+    @classmethod
+    def calibrationGroupingHome(cls) -> Path:
+        return Path(Config["instrument.calibration.powder.grouping.home"])
+
+    # Use the StateId hash to enforce filesystem-as-database integrity requirements:
+    # * verify that this GroupingMap's file is at its expected location (e.g. it hasn't been moved or copied);
+    stateId: ObjectSHA
+
+    nativeFocusGroups: List[FocusGroup] = Field(default=None)
+    liteFocusGroups: List[FocusGroup] = Field(default=None)
+
+    _isDirty: bool = False
+    _nativeMap: Dict[str, FocusGroup] = None
+    _liteMap: Dict[str, FocusGroup] = None
+
+    @property
+    def lite(self) -> Dict[str, FocusGroup]:
+        return self._liteMap
+
+    @property
+    def native(self) -> Dict[str, FocusGroup]:
+        return self._nativeMap
+
+    @property
+    def isDefault(self):
+        return self.stateId == GroupingMap.defaultStateId
+
+    @property
+    def isDirty(self) -> bool:
+        return self._isDirty
```

Comment:
I don't think that "dirty" applies to groupings, as they should never be edited within SNAPRed for any reason.

Reply:
`GroupingMap` is not a <grouping>.  The `GroupingMap` itself is "dirty" if its `stateId` has been modified from the default.  In this case, it must be written out to <state root>.  This only happens at initialization of a new state version tree, and the `GroupingMap` is only written once, and never allowed to be overwritten.